client/app/models/event.coffee

```diff
@@ -101,6 +101,81 @@ module.exports = class Event extends ScheduleItem
     getDefaultColor: -> '#008AF6'
 
 
+    # A shared event has at least one attendees with a cozy invitation
+    isShared: ->
+        attendees = @get 'attendees'
+        cozyAttendees = attendees?.find (attendee) ->
+            return attendee.shareWithCozy
+        return cozyAttendees?
+
+
+    # Try to get a shareID (called typically just after saving),
+    tryGetShareID: (numtries, delay, callback) ->
+
+        @fetch
+            success: (model, response) =>
+                shareID = model.get 'shareID'
+                if shareID
+                    callback null, shareID
+                    return
+
+                triesLeft = --numtries;
+
+                if numtries
+                    setTimeout =>
+                        @tryGetShareID triesLeft, delay, callback
+                    , delay
+                else
+                    callback 'Could not retrieve shareID, maximum \
+                        number of tries exceeded',
+                        null
+            error: (model, response) ->
+                callback(response.error or response, null)
+
+
+    # Fetch the sharing object to handle any error
+    onShareIDChange: ->
+        @fetchSharing (err, sharing) =>
+            if err
+                throw
+                    name: 'EventSharingError',
+                    event: @
+                    message: 'cannot retrieve sharing : ' + err
```

Comment:
I think `'cannot retrieve sharing : '` is useless because you already typed the error.

You're right, translating can be done when the string is displayed.